### PR TITLE
feat(query): insert/mutation/update deduplicate based on label

### PR DESCRIPTION
**Pull request by jun0315**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

insert/mutation/update based on label

### TODO
wait for copy hint:https://github.com/datafuselabs/databend/pull/11614

Closes #https://github.com/datafuselabs/databend/issues/11591

### CODE REVIEW

File: src/query/settings/src/settings_default.rs [hunk 1]
```diff
@@ -310,6 +310,12 @@ impl DefaultSettings {
                     possible_values: None,
                     display_in_show_settings: true,
                 }),
+                ("duplicate_label", DefaultSettingValue {
```
Comment (sundy-li) [inline src/query/settings/src/settings_default.rs:313]: better name as deduplicate_label?

File: src/query/storages/fuse/src/operations/util.rs [hunk 2]
```diff
@@ -73,3 +82,39 @@ pub fn column_parquet_metas(
...
+    if duplicate_label.is_empty() {
+        Ok(false)
+    } else {
+        let kv_store = UserApiProvider::instance().get_meta_store_client();
+        let raw = kv_store.get_kv(&duplicate_label).await?;
+        match raw {
+            None => {
+                let expire_at = SeqV::<()>::now_ms() / 1000 + 24 * 60 * 60 * 1000;
+                let _ = kv_store
+                    .upsert_kv(UpsertKV {
```
Comment (sundy-li) [inline src/query/storages/fuse/src/operations/util.rs:106]: check should be checked before the whole query pipeline.

upsert should be committed with the update meta commit.

File: src/meta/api/src/schema_api_impl.rs [hunk 3]
```diff
@@ -2379,6 +2381,12 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
                 txn_req.if_then.extend(match_operations)
             }
 
+            if let Some(deduplicated_label) = req.deduplicated_label.clone() {
+                txn_req
+                    .if_then
+                    .push(build_upsert_table_deduplicated_label(deduplicated_label))
+            }
+
```
Comment (drmingdrmer) [inline src/meta/api/src/schema_api_impl.rs:2389]: Who will consume this label? 

I assume every SchemaApi method that updates the copied file must assert its label does not exists to proceed.

File: src/meta/api/src/schema_api_impl.rs [hunk 4]
```diff
@@ -3230,6 +3238,18 @@ fn build_upsert_table_copied_file_info_conditions(
     Ok((condition, if_then))
 }
 
+fn build_upsert_table_deduplicated_label(deduplicated_label: String) -> TxnOp {
+    let expire_at = Some(SeqV::<()>::now_ms() / 1000 + 24 * 60 * 60 * 1000);
+    TxnOp {
+        request: Some(Request::Put(TxnPutRequest {
```
Comment (drmingdrmer) [inline src/meta/api/src/schema_api_impl.rs:3244]: 1000 days is a little bit long, so there should be a mechanism to clean them up.

**Review by drmingdrmer — APPROVED** [review]

**Review by sundy-li — APPROVED** [review]

File: tests/suites/0_stateless/05_hints/05_0002_deduplicate_label.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+echo "INSERT /*+ SET_VAR(deduplicate_label='insert-test') */ INTO t1 (a, b) VALUES(1, false)" | $MYSQL_CLIENT_CONNECT
+echo "INSERT /*+ SET_VAR(deduplicate_label='insert-test') */ INTO t1 (a, b) VALUES(1, false)" | $MYSQL_CLIENT_CONNECT
+echo "select * from t1" | $MYSQL_CLIENT_CONNECT
+
+echo "CREATE STAGE s0;" | $MYSQL_CLIENT_CONNECT
+echo "copy into /*+SET_VAR(deduplicate_label='copy-test')*/ @s0 from (select * from t1);" | $MYSQL_CLINEENRT_CONNECT
+echo "select * from @s0;" | $MYSQL_CLINEENRT_CONNECT
+echo "CREATE STAGE s1;" | $MYSQL_CLINEENRT_CONNECT
+echo "copy into /*+SET_VAR(deduplicate_label='copy-test')*/ @s1 from (select * from t1);" | $MYSQL_CLINEENRT_CONNECT
+echo "select * from @s0;" | $MYSQL_CLINEENRT_CONNECT
```
Comment (soyeric128) [inline tests/suites/0_stateless/05_hints/05_0002_deduplicate_label.sh:20]: `select * from @s1`?

File: tests/suites/0_stateless/05_hints/05_0002_deduplicate_label.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+echo "UPDATE /*+ SET_VAR(deduplicate_label='update-test') */ t SET a = 20 WHERE b = false;" | $MYSQL_CLIENT_CONNECT
```
Comment (soyeric128) [inline tests/suites/0_stateless/05_hints/05_0002_deduplicate_label.sh:22]: t1?

File: tests/suites/0_stateless/05_hints/05_0002_deduplicate_label.sh [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+# mysql --version
+# mysql  Ver 8.0.32-0ubuntu0.20.04.2 for Linux on x86_64 ((Ubuntu))
+echo "CREATE TABLE t1(a Int, b bool) Engine = Fuse;" |  $MYSQL_CLIENT_CONNECT
+
+echo "INSERT /*+ SET_VAR(deduplicate_label='insert-test') */ INTO t1 (a, b) VALUES(1, false)" | $MYSQL_CLIENT_CONNECT
+echo "INSERT /*+ SET_VAR(deduplicate_label='insert-test') */ INTO t1 (a, b) VALUES(1, false)" | $MYSQL_CLIENT_CONNECT
+echo "select * from t1" | $MYSQL_CLIENT_CONNECT
+
+echo "CREATE STAGE s0;" | $MYSQL_CLIENT_CONNECT
+echo "copy into /*+SET_VAR(deduplicate_label='copy-test')*/ @s0 from (select * from t1);" | $MYSQL_CLINEENRT_CONNECT
```
Comment (soyeric128) [inline tests/suites/0_stateless/05_hints/05_0002_deduplicate_label.sh:16]: It seems set_var must be placed between COPY and INTO to make it work

**Pull request merged**